### ISSUE TITLE
Outer right border radius not applied to datalist in an input group

### ISSUE THREAD

**Issue by omishah** [twbs/bootstrap#40763]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

**Issue**: Outer top-right and bottom-right border radius not applied to input (datalist) when used with the `input-group` class.

**Inshort**: `border-top-right-radius` and `border-bottom-right-radius` properties are set to 0 instead of default value of `--bs-border-radius`.

Code:

```
<div class="input-group">
    <span class="input-group-text">#</span>

    <input list="datalistOptions" class="form-control datalist" placeholder="Choose or type an option">

    <datalist id="datalistOptions">
        <option value="abc">abc</option>
        <option value="...">...</option>
        <option value="xyz">xyz</option>
    </datalist>
</div>
```

**Preview:**

<img width="394" alt="Screenshot 2024-08-22 at 21 31 44" src="https://github.com/user-attachments/assets/90abf21b-23f9-42a8-98ba-e2f9329f4fe4">

### Reduced test cases

**Fix:**

When manually setting respective radius, it works fine:

```
input.datalist {
    border-top-right-radius: var(--bs-border-radius) !important;
    border-bottom-right-radius: var(--bs-border-radius) !important;
}
```

**Preview:**

<img width="389" alt="Screenshot 2024-08-22 at 21 38 28" src="https://github.com/user-attachments/assets/edc89bfe-fef1-41b0-ab0f-e865362dcd35">

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

v5.3.3

**Comment by TommasoAllegretti** [comment]
With input groups, the border radius style is applied to the first and last children of the element with `class="input-group"`.

In your case the border radius is not displayed correctly because the last child element is actually `<datalist>` and not `<input>`, changing the order of those two tags should solve the issue.

**Comment by omishah** [comment]
> With input groups, the border radius style is applied to the first and last children of the element with `class="input-group"`.
> 
> In your case the border radius is not displayed correctly because the last child element is actually `<datalist>` and not `<input>`, changing the order of those two tags should solve the issue.

Yes, it makes sense i.e changing the order works. But, I guess it's worth mentioning in the official docs. Thanks.

**Comment by TommasoAllegretti** [comment]
I agree. I will create a PR to add it to the documentation.
